src/dialects/db2/query-generator-typescript.ts

```diff
@@ -33,4 +36,26 @@ export class Db2QueryGeneratorTypeScript extends AbstractQueryGenerator {
       'ORDER BY NAME;',
     ]);
   }
+
+  removeIndexQuery(tableName: TableNameOrModel, indexNameOrAttributes: string | string[], options: RemoveIndexQueryOptions) {
+    if (options) {
+      rejectInvalidOptions(
+        'removeIndexQuery',
+        this.dialect.name,
+        REMOVE_INDEX_QUERY_SUPPORTABLE_OPTIONS,
+        new Set(),
+        options,
+      );
+    }
+
+    let indexName: string;
+    const table = this.extractTableDetails(tableName);
+    if (Array.isArray(indexNameOrAttributes)) {
+      indexName = generateIndexName(table, { fields: indexNameOrAttributes });
```

Comment:
Would it maybe be useful to move this checking for an array to `generateIndexName`? So we can remove this check in all dialects and implement it once in `generateIndexName`

Reply:
If the value is not an array, I am assuming that the index name was passed to the function and not a field.
If the index name is passed to the `generateIndexName` we would end up with an invalid index.

Not sure if this is the best solution however this is just the previous implementation of the function

Reply:
This should be fine, we can keep this